draft-west-cookie-incrementalism.md

```diff
@@ -401,85 +404,361 @@ from "have the same registrable domain" (or "is an exact match for") to say
+   which allows users to use the services. To get a token a user logs
+   in on `https://corp.example/login/` which creates the cookie
+   `IDToken=a1b2bc3`. Next, to file an expense, the user needs to
+   visit `https://corp.example:8443`, but when they do the IDToken
+   isn't sent because the ports differ (443 vs 8443) and the user is
+   denied access. To remedy this, the IDToken can have the `Domain` 
+   attribute added to it: `IDToken=a1b2d3; Domain=corp.example`.
+   Now when the user visits `https://corp.example:8443` the token
+   is sent and access is granted as expected.
+
+Because any domain cookie is now exposed to multiple origins it means
+a cookie created by one origin can be overwritten by another origin. 
+
+Examples:
+
+1. `https://example.com:123` sets a cookie: `Set-Cookie: 
+    foo=domaincookie; Domain=example.com`. The user then visits 
+    `https://example.com:456` which sets the same cookie. This second
+    cookie would match the first and then overwrite it. 
+2. `https://example.com:123` sets a cookie 
+    `Set-Cookie: foo=domaincookie; Domain=example.com`. The user then
+    visits `https://example.com:456` which sets `Set-Cookie: 
+    foo=origincookie`. The second cookie would match the first an
+    overwrite it.
+    
+Also, because domain cookies are less trusted due to their wider scope,
+we'll want to avoid allowing them to shadow non-domain cookies by
+disallowing domain cookies from co-exiting or overwriting a matching
+non-domain cookie. This is a departure from the status quo in which
+this shadowing behavior is specifically allowed.
+
+Example:
 
-Cookies are one of the very few components of the web platform that ignore scheme by default. The
-`Secure` attribute can lock a cookie to secure schemes, and the `__Secure-` prefix can harden that
-boundary, but these mechanisms are little-used, and cookies lacking these protections flow across
-scheme boundaries. They are delivered to both the HTTP and HTTPS variants of a given domain, even
-though their security properties differ radically. As Section 8.6 of
-{{RFC6265bis}} points out, this gives network attackers the ability to influence otherwise secured
-traffic by modifying user state that flows to secure origins, and, of course, insight into user
-behavior as securely-set cookies that lack the `Secure` attribute likewise flow from secure origins
-to non-secure variants.
+* `https://example.com:456` sets a non-domain cookie`Set-Cookie:
+   foo=origincookie`. The user then visits `https://example.com:123`
+   which tries to set a domain cookie `Set-Cookie: foo=domaincookie;
+   Domain=example.com` This set fails because a domain cookie should
+   never overwrite a matching non-domain cookie.
+
+Finally, we don't ever want to allow a cookie to pass between schemes,
+given the huge security differences between them. So there should be no
+way for a server to specify that a given cookie should be sent to a
+different scheme.
```

Comment:
As in one of the other comment threads, this is going to be a big change in itself, and it might make sense for you to structure the proposal as explicitly staged.